src/Azure.Functions.Cli/Actions/HostActions/StartHostAction.cs

```diff
@@ -418,15 +429,104 @@ private async Task<JObject> GetLocalSettingsJsonAsJObjectAsync()
         public override async Task RunAsync()
         {
             await PreRunConditions();
+            var isVerbose = VerboseLogging.HasValue && VerboseLogging.Value;
 
             var isCurrentProcessNet6Build = RuntimeInformation.FrameworkDescription.Contains(Net6FrameworkDescriptionPrefix);
-            if (isCurrentProcessNet6Build && ShouldLaunchInProcNet8AsChildProcess() && await IsInProcNet8Enabled())
+            if (SetHostRuntime != null)
```

Comment:
This method has become too large, affecting its readability. Could we refactor it by extracting parts into smaller, more manageable methods?

Reply:
done